internal/docs/design.md

```diff
@@ -96,6 +96,14 @@ https://godoc.org/github.com/google/go-cloud/runtimevar#Variable
 https://godoc.org/github.com/google/go-cloud/blob#Bucket.NewWriter
 [`database/sql`]: https://godoc.org/database/sql
 
+## No Global State
+
+As a library, Go Cloud should not introduce global state. Global state is
```

Comment:
I think this advice is too strict. We already [violate it](https://github.com/google/go-cloud/blob/master/blob/blob.go#L618). 

Maybe no exported global state? Still too strict for me, actually.

Reply:
I guess I was thinking of globals that one sometimes writes instead of declaring flags inside a package.

Anyway, I just wanted to record my thoughts. I won't block this.